### PR TITLE
Support for tagging rating and review values

### LINKED ISSUE #50: Add support for review counts and rating values

### PR DESCRIPTION
**Pull request by mridang**
Closes #50

### CODE REVIEW

File: Model/Product/Builder.php [hunk 1]
```diff
@@ -135,6 +142,42 @@ public function build(Product $product, StoreInterface $store)
...
+     * @return float the normalized rating value of the product
+     */
+    private function buildRatingValue(Product $product, StoreInterface $store) {
+        /** @noinspection PhpUndefinedMethodInspection */
+        if (!$product->getRatingSummary()) {
+            $this->reviewFactory->create()->getEntitySummary($product, $store->getId());
+        }
+
+        /** @noinspection PhpUndefinedMethodInspection */
+        return round($product->getRatingSummary()->getRatingSummary() / 20, 1);
```
Comment (oostman) [inline Model/Product/Builder.php]: Can ratingSummary be null or 0?

File: Model/Product/Builder.php [hunk 2]
```diff
@@ -135,6 +142,42 @@ public function build(Product $product, StoreInterface $store)
...
+     * normalised to a 0-5 value.
+     *
+     * @param Product $product the product whose rating value to fetch
+     * @param StoreInterface $store the store scope in which to fetch the rating
+     * @return float the normalized rating value of the product
+     */
+    private function buildRatingValue(Product $product, StoreInterface $store) {
+        /** @noinspection PhpUndefinedMethodInspection */
+        if (!$product->getRatingSummary()) {
+            $this->reviewFactory->create()->getEntitySummary($product, $store->getId());
```
Comment (oostman) [inline Model/Product/Builder.php]: Are we creating an empty entry with reviewFactory->create()? Is it stored to database? Why?

File: Model/Product/Builder.php (hunk 1 above)
Comment (hannunosto) [inline Model/Product/Builder.php]: I would add also add safeguard for checking null & 0

File: Model/Product/Builder.php (hunk 2 above)
Comment (hannunosto) [inline Model/Product/Builder.php]: @oostman not really creating an empty entry. This is just following M2 factory pattern where you don't instantiate objects directly but use factory.

File: Model/Product/Builder.php (hunk 1 above)
Comment (mridang) [inline Model/Product/Builder.php]: I looked at how Magento is doing it in the ReviewRenderer and they weren't doing any null checks. Also, it seems to work fine when the product doesn't have any ratings or reviews.

File: Model/Product/Builder.php (hunk 2 above)
Comment (mridang) [inline Model/Product/Builder.php]: That's just the way you're supposed to do it. https://github.com/magento/magento2/blob/2.1.5/app/code/Magento/Review/Block/Product/ReviewRenderer.php#L72

**Pull request merged**